src/shogun/features/RandomFourierDotFeatures.cpp

```diff
@@ -83,27 +83,27 @@ float64_t RandomFourierDotFeatures::post_dot(float64_t dot_result, index_t par_i
 	return std::cos(dot_result) * constant;
 }
 
-SGVector<float64_t> RandomFourierDotFeatures::generate_random_parameter_vector()
+void RandomFourierDotFeatures::generate_random_parameter_vector(SGMatrix<float64_t>& mat, index_t col)
 {
 	NormalDistribution<float64_t> normal_dist;
 	UniformRealDistribution<float64_t> uniform_real_dist(0.0, 2 * Math::PI);
-	SGVector<float64_t> vec(feats->get_dim_feature_space()+1);
+	auto len = mat.num_rows + 1;       
+
 	switch (kernel)
 	{
 		case GAUSSIAN:
```

Comment:
@karlnapf is there a reason why we can’t reuse the kernel implementation of a Kernel class?

Reply:
For example GaussianKernel in this case

Reply:
The code to sample spectral densities of the kernel (happens to be a gaussian for the gaussian kernel) doesnt really exist in the kernel class, but I agree we should add it there. The OOP way to do this would be to add a virtual method `sample_spectral_density` to `Kernel`, and then throw a runtime error if that is not implemented. Then all this switching here would be unnecessary. Furthermore, the shitty passing the kernel bandwidth to the embedding class would not happen (instead it would just read it from the kernel object). I have suggested that to @NanuSai already.

Reply:
@NanuSai in my opinion this should be next (after we have merged this)